### PR TITLE
Clean up `search_query_location` code.

### PR DESCRIPTION
**Pull request by hblumberg**
As I'm trying to better understand how `app.js` works, I'm doing small clean ups! Mostly moving utility functions to a separate file and deleting dead code here.

### Test Plan:
Navigate to `/search` and enter a few ZIP codes. Ensure 5 results are consistently returned.

Screen Shot 2021-02-10 at 5 16 53 PM

Check the console log and confirm there are no errors.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -170,77 +173,25 @@ if (cluster.isMaster) {
...
-        var indices = findIndicesOfMax(distance_array, 5);
-        
-
-        res.send(indices);
-    });
-
-    var port = process.env.PORT || 3002;
-
-    var server = app.listen(port, function () {
-        console.log('Server running at http://127.0.0.1:' + port + '/');
```
Comment (hblumberg) [inline app.js]: I think this is dead code?

File: utils/distance-utils.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+function degreesToRadians(degrees) {
```
Comment (hblumberg) [inline utils/distance-utils.js:1]: The functions here are all copied from `app.js` (with some naming changes in `getClosestLocations`, previously `findIndicesOfMax`).

File: app.js (hunk 1 above)
Comment (johndeyrup) [reply app.js]: I think this is how you start the server?

File: utils/distance-utils.js (hunk 2 above)
Comment (johndeyrup) [reply utils/distance-utils.js:1]: :+1: like the util methods

File: app.js [hunk 3]
```diff
@@ -170,77 +173,25 @@ if (cluster.isMaster) {
...
-                    outp.pop(); // remove the last index (index of smallest element in output array)
-                }
-            }
-            var closest = []
-            outp.forEach(function(element) {
-                closest.push(locations[element])
-            })
-            return closest;
+        var distanceArray = [];
+        for (var i = 0; i < locations.length; i++) {
```
Comment (johndeyrup) [inline app.js]: might be a little clearer to add the distance to each location and then sort the location array by distance and take the first to nth

Comment (hblumberg) [reply app.js]: Agreed, will do!

I was wondering if it was done this way (i.e. avoiding a full `sort`) for performance reasons though.. maybe @zzany can speak to that?

File: app.js (hunk 1 above)
Comment (hblumberg) [reply app.js]: 🤦 Restored!

File: app.js (hunk 3 above)
Comment (johndeyrup) [reply app.js]: There are only 140 sites so performance isn't that big of a deal either way and I don't see it ever being a big deal. I would be shocked and very happy if MA set up 10,000 vaccination sites. The current algorithm doesn't seem to be more performant than a single sort. It iterates through the distances and adds them to an array of size n, once the n distance array is full it adds one more element resorts the array and pops off the last one to get back to size n. That is less efficient than doing a single sort; although it saves space I suppose.

**Review by johndeyrup — APPROVED** [review]

**Comment by zzany** [comment]
Actually can we revert on this? Looks like it broke the "search near my
location" function

**Comment by zzany** [comment]
Nvm my mistake

**Pull request merged**